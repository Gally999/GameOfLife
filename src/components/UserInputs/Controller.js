import React from 'react';
import PropTypes from "prop-types";

import Options from './Options';
import Player from '../UserInputs/Player';

const Controller = ({
	x,
	y,
	steps,
	onDimensionsChange,
	onEvolveClick,
	onStepsUpdate,
}) => (
	<div>
		<Options
			x={x}
			y={y}
			onDimensionsChange={onDimensionsChange}
		/>
		<Player
			steps={steps}
			onEvolveClick={onEvolveClick}
			onStepsUpdate={onStepsUpdate}
		/>
	</div>
);

Controller.propTypes = {
	x: PropTypes.number.isRequired,
	y: PropTypes.number.isRequired,
	steps: PropTypes.number.isRequired,
	onDimensionsChange: PropTypes.func.isRequired,
	onEvolveClick: PropTypes.func.isRequired,
	onStepsUpdate: PropTypes.func.isRequired,
}

export default Controller;
