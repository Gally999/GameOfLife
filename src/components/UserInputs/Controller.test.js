import React from 'react';
import Controller from './Controller';

describe('Controller', () => {
  it('renders the controller component correctly', () => {
    const subject = (
      <Controller
        x={4}
        y={7}
        steps={1}
        onDimensionsChange={() => {}}
        onEvolveClick={() => {}}
        onStepsUpdate={() => {}}
      />
    );
    expect(subject).toMatchSnapshot();
  });
});
