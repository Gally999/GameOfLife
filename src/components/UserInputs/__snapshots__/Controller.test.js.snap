// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller renders the controller component correctly 1`] = `
<Controller
  onDimensionsChange={[Function]}
  onEvolveClick={[Function]}
  onStepsUpdate={[Function]}
  steps={1}
  x={4}
  y={7}
/>
`;
